### ISSUE TITLE
Configurei minha própria infra com Asterisk e um SIP Trunk

### ISSUE THREAD

**Issue by rafaelcorreiapoli** [haskellcamargo/gemidao-do-zap#61]
Pessoal, acho que já era o TotalVoice
Passei a última madrugada configurando minha própria infra e cheguei em algo bem legal, segui as idéias apresentadas aqui:
https://www.youtube.com/watch?v=DZ0czppbamo&t=189s
Subi o server de asterisk com docker
https://github.com/dougbtv/docker-asterisk

Criei um aplicativo Android/iOS com react native que utiliza essa minha infra pra fazer o Gemidão (igualzinho esta CLI, só que com uma interface gráfica)
http://appgemidaodozap.com.br/

Asterisk: http://www.asterisk.org/
FreePBX: https://www.freepbx.org/
Flowroute: https://www.flowroute.com/

To dando de graça o primeiro gemidão e depois tenho que cobrar para pagar os custos do SIP trunk, servidores, meu tempo, etc...

**Comment by haskellcamargo** [comment]
:heart: não deixa o gemidão morrer!

**Comment by luizlago** [comment]
Comentei lá na play store e vou comentar aqui tb: 

Alguns feedbacks: 
- Ele desconta no credito mesmo se a ligação não completar(por exemplo, se esquecer de digitar o 9 na frente do celular vc perde o crédito mesmo assim) 

- O custo está muito alto, 4,99 por 3 utilizações não vale a pena...acredito que dessa forma vc vai ganhar muitos usuários (vão criar varias contas para receber o primeiro gratis) e poucos pagantes. Poderia resolver isso se ajustasse o preço para algo mais equilibrado, tipo 3 utilizações = 2 reais...acho que isso seria justo.

**Comment by rafaelcorreiapoli** [comment]
@luizlago Valeu pelas dicas! o problema 1 já está sendo resolvido
Inclusive implementei ontem tracking da ligação em tempo real (socket.io FTW), vc consegue acompanhar quando o telefone da pessoa começa a tocar, quando ela atende e desliga

Sobre o custo.. eu to tendo um prejuízo absurdo por causa do CAC (Customer acquisition cost), que é exatamente esse gemidão grátis que eu dou pro user e, do jeito que tá, tá insustentável

Como to pagando o serviço em dólar, cada gemidão sai por quase R$ 0,80 centavos, então esse valor que vc falou é inviável pois tem os custos de transação do PagSeguro

Além disso, a App Store me esculachou e disse que se eu tentar enviar um app desse de novo eles vão cancelar minha conta

Enfim.. to quase desistindo do negócio, apesar de enxergar um enorme potencial
Se alguém ai tiver interesse em montar um plano de negócios e estratégia legal e ajudar a administrar o app, entre em contato :P

**Comment by teteucalifornia2** [comment]
Aqui não veio o gemidão gratuito pra testar

**Comment by rafaelcorreiapoli** [comment]
@teteucalifornia2 Eu removi o gemidão gratuíto e abaixei o preço dos pacotes
Esse gemidão gratuíto tava pesando no meu bolso devido ao grande número de usuários

**Comment by teteucalifornia2** [comment]
@rafaelcorreiapoli Entendo. O foda é como as pessoas vão comprar sem saber se realmente funciona. Porém, por outro lado se todo mundo usar só o gratuito, você pode sair num prejuízo monstro.

**Comment by rafaelcorreiapoli** [comment]
Sim.. é um tradeoff
Estou pensando em gravar um vídeo monstrando que funciona e colocar no site ou mesmo no app, da primeira vez que a pessoa abre
Por enquanto, estou contando com duas formas de disseminação:
No boca a boca, ou seja, a pessoa vai acreditar que funciona pois um amigo irá indicar e
Implementei um sistema de disparo de SMS para a pessoa trollada após 15 min da trollagem, falando sobre a brincadeira e com um link pra baixar o app. (Esta pessoa irá acreditar pois, afinal, ela foi vítima)

**Comment by teteucalifornia2** [comment]
São boas soluções, realmente. Amigo meu disse outro dia que pagaria pra ''trollar'' os amigos, vou passar pra ele e vejo se ele anima testar mesmo.

**Comment by luizsaldanha** [comment]
Rafael, seu projeto APK é opensource? Tem como configura-lo para funcionar com o TotalVoice para contas que ainda estão ativas?

**Comment by rafaelcorreiapoli** [comment]
Não está open source não
E essa configuração é no servidor, não é no App. Mas é relativamente fácil de colocar a opção da conta usar créditos do TotalVoice e não do meu sistema
Quanto de crédito vc tem lá?

**Comment by luizsaldanha** [comment]
Aprox 60,00. Estão liberando contas pra CNPJ primeiro, acredito que quem tiver conta como PF vão demorar mais...

**Comment by paulopwm** [comment]
Consegue disponibilizar a configuração que tu fez no asterisk?

**Comment by luizsaldanha** [comment]
Olhem esse APP que está usando a API do TotalVoice, https://play.google.com/store/apps/details?id=com.renanrhoden.gemidaocall&hl=en 
suspeito ou não essas permissões??? 
-
Gemidão Call (Ligacao do gemidao do zap)
Renan Rhoden
read your contacts
approximate location (network-based)
precise location (GPS and network-based)
read the contents of your USB storage
modify or delete the contents of your USB storage
take pictures and videos
record audio
manage document storage
receive data from Internet
view network connections
full network access
draw over other apps
control vibration
prevent device from sleeping
install shortcuts
read Google service configuration

**Comment by rafaelcorreiapoli** [comment]
@paulopwm Fiz seguindo as idéias do vídeo
Infelizmente não posso disponibilizar

**Comment by rafaelcorreiapoli** [comment]
@luizsaldanha Me chama lá no facebook pra gente conversar? https://www.facebook.com/rafael.ribeirocorreia.9
Abraços!

**Comment by luizsaldanha** [comment]
TotalVoice it's over... Agora não terá mais spoof, tem cadastrar os números que vão ligar

**Comment by rafaelcorreiapoli** [comment]
Alguém sabe como esse aqui https://www.troll4me.com/ tá funcionando ?

**Comment by walterdf** [comment]
@rafaelcorreiapoli  nem idéia.. sua configuração não funciona mais?

**Comment by Vagson** [comment]
Esse site https://www.troll4me.com/ tá funcionando perfeitamente alguém sabe como? qual a a plataforma que estão usando?

**Comment by jhonata-menezes** [comment]
No facebook tem um comentário do CEO da TotalVoice informando que eles irão deixar apenas números que forem confirmados aparecerem na bina.

**Comment by Skolzack** [comment]
Nego é maluco de colocar CC nesse troll4me, pqp

**Comment by williamokano** [comment]
Uai, eu coloco de boa, só gerar cartão virtual

**Comment by CesarSantosBR** [comment]
@rafaelcorreiapoli, conseguiu verificar se consegue a parte abaixo:

"Não está open source não
E essa configuração é no servidor, não é no App. Mas é relativamente fácil de colocar a opção da conta usar créditos do TotalVoice e não do meu sistema
Quanto de crédito vc tem lá?"

Ainda tenho R$ 15 no Total voice =/

**Comment by GabrielCoelho** [comment]
Eu tenho R$ 6 lá no totalvoice totalmente parado tambem. 

Não deixe o gemidao morrer 

_Sent from my Samsung SM-G532MT using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by rafaelcorreiapoli** [comment]
Galera, agora não adianta mais os créditos no TotalVoice
eles bloquearam o envio a partir de números não cadastrados
Independente do meu interesse, achei isso sacanagem com quem comprou crédito antes da mudança... a pessoa pagou por uma coisa e recebeu outra...

**Comment by jhonata-menezes** [comment]
Rafael sem falar que eles não comunicaram os clientes e nem deram tempo para para realizar ajustes. Como cliente a totalvoice perde muita credibilidade.

**Comment by lucastex** [comment]
@jhonata-menezes Se não me engano, só mudaram esse procedimento para as contas novas, criadas na era "pós-gemidão-do-zap" hahhahahha

**Comment by jhonata-menezes** [comment]
@lucastex realmente, minha conta continua a fazer ligação normalmente com qualquer numero no Bina.

**Comment by lucastex** [comment]
@jhonata-menezes sim :) eu tb sou parceiro deles e continuo usando tudo normalmente...

**Comment by CesarSantosBR** [comment]
Então só eu que me ferrei nessa história?

**Comment by williamokano** [comment]
Não, muita gente se ferrou rs. Eu mesmo estou com 18 reais parados lá.

**Comment by henriqueweis** [comment]
Rafael, 
Tinha q desabilitar o SMS. 
Como q faz?
E qndo não sai do preparando ligação, gasta crédito mesmo assim

**Comment by rafaelcorreiapoli** [comment]
Olá @henriqueweis 
Por enquanto não tem como desabilitar o SMS
Faz parte da "segurança" da brincadeira avisar depois de um tempo que foi uma brincadeira... já teve casos de pessoas mais velhas que não entenderam a brincadeira e acharam que era sequestro (sim, podem dar risada mas é verdade) e por isso implementei essa política
O tratamento de erros (e a consequente devolução dos créditos) ainda é algo que tem que ser melhorado.
De qualquer maneira, eu só desconto o crédito se o serviço de telefonia me responde a requisição com um OK. A partir dai, se a pessoa atendeu ou não, ou se realmente funcionou ou não, não importa. Será cobrado da mesma maneira pois o serviço de telefonia já me cobrou pela ligação

**Comment by henriqueweis** [comment]
@rafaelcorreiapoli 
esse negocio de logar com o facebook que tambem podia melhorar.
tenho 2 facebooks e preciso toda hora deslogar de um para poder logar no app.
